MTA-3588: Release Notes MTA 7.2.0 (#67)
migtools/mta-documentation: docs/topics/ref_known-issues-7-2-0.adoc

:_newdoc-version: 2.18.3
:_template-generated: 2024-08-14

:_mod-docs-content-type: REFERENCE

[id="known-issues-7-2-0_{context}"]
= Known issues

This section provides highlighted known issues in {ProductFullName} version 7.2.0.

> .Enabling Preemption does not work if authentication is enabled
> 
> In {ProductFullName} 7.1.0, *Preemption* cannot be enabled for tasks in the Task Manager page and the Task Manager drawer if {ProductShortName} authentication is enabled. To work around this issue, disable authentication. link:https://issues.redhat.com/browse/MTA-3195[(MTA-3195)]
> 

Author: ```suggestion
```

> .CLI analysis works extremely slow on Windows

Author: @ibragins - MTA-3667 added to the release notes. Is there anything else you would like me to add?
Thanks

Reviewer: thank you, that is it

> {ProductShortName} CLI analysis does not function as expected on Microsoft Windows, with analysis using container runtime being extremely slow on Windows. This issue should be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-3667[(MTA-3667)]

Author: ```suggestion
{ProductShortName} CLI analysis does not function as expected on Microsoft Windows, with analysis using container runtimes being extremely slow on Windows. This issue should be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-3667[(MTA-3667)]
```

:_template-generated: 2024-12-04
:_mod-docs-content-type: REFERENCE

[id="known-issues-7-2-0_{context}"]
= Known issues

This section provides highlighted known issues in {ProductFullName} version 7.2.0.

.CLI analysis is extremely slow on Windows

> {ProductShortName} CLI analysis does not function as expected on Microsoft Windows, with analysis using container runtimes being extremely slow on Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-3667[(MTA-3667)]

Author: @ibragins - FYI

> .Analysis of the daytrader app does not show dependencies

Author: ```suggestion
.Analysis of the daytrader application does not show dependencies
```

> The analysis of the daytrader app does not show dependencies. This failure to display dependencies is caused by the {ProductShortName} Operator using Maven 3.6, which sometimes fails to fetch dependencies. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]

Reviewer 2: Nitpick: Could app be replaced with application ?

> The analysis of the daytrader app does not show dependencies. This failure to display dependencies is caused by the {ProductShortName} Operator using Maven 3.6, which sometimes fails to fetch dependencies. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]

Author: ```suggestion
The analysis of the daytrader application does not show dependencies. This failure to display dependencies is caused by the {ProductShortName} Operator using Maven 3.6, which sometimes fails to fetch dependencies. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]
```

.CLI bulk analysis does not function on Windows

> {ProductShortName} CLI bulk analysis, performed by using the `--bulk` option, does not function on Microsoft Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4307[(MTA-4307)]

Author: @ibragins - FYI

> Running analysis of the application in containerless mode fails to fetch some dependencies, which is caused by the {ProductShortName} Operator using Maven 3.6 that can sometimes fails to fetch dependencies. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]

Author: ```suggestion
Running analysis of the application in containerless mode fails to fetch some dependencies, which is caused by the {ProductShortName} Operator using Maven 3.6 is failing to fetch dependencies. The workaround is for the user to upgrade to maven 3.9. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]
```

:_template-generated: 2024-12-04
:_mod-docs-content-type: REFERENCE

[id="known-issues-7-2-0_{context}"]
= Known issues

This section provides highlighted known issues in {ProductFullName} version 7.2.0.

> .{ProductShortName} CLI analysis using the CLI in container and containerless modes periodically fails on Windows

Author: @istein1 - done

> .{ProductShortName} CLI analysis using analysis using the CLI in container and containerless modes periodically fails on Windows

Author: @ibragins - FYI. thanks

> .{ProductShortName} CLI analysis using analysis using the CLI in container and containerless modes periodically fails on Windows

Author: ```suggestion
.{ProductShortName} CLI analysis using the CLI in container and containerless modes periodically fails on Windows
```

> {ProductShortName} [CLI] analysis using the CLI in container and containerless modes periodically fails on windows on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of a corrupted `.metadata` folder located inside the directory where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]

Author: ```suggestion
{ProductShortName} [CLI] analysis using the CLI in container and containerless modes periodically fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of a corrupted `.metadata` folder located inside the directory where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]
```

.Running analysis of the application in containerless mode fails to fetch some dependencies

Running analysis of the application in containerless mode fails to fetch some dependencies, which is caused by the {ProductShortName} Operator using Maven 3.6 is failing to fetch dependencies. The workaround is for the user to upgrade to Maven 3.9. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]

.{ProductShortName} CLI bulk analysis does not function on Windows

{ProductShortName} CLI bulk analysis, performed by using the `--bulk` option, does not function on Microsoft Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4307[(MTA-4307)]

.Binary analysis fails with incorrect Maven credentials

{ProductShortName} CLI analysis fails with a generic error when incorrect Maven credentials are used. The failure occurs because the analyzer is unable to report detailed error messages. link:https://issues.redhat.com/browse/MTA-3714[(MTA-3714)]

> .{ProductShortName} containerless CLI report is inconsistent when analyzing two binary applications located in the same folder

Author: @sshveta - [MTA-4484](https://issues.redhat.com/browse/MTA-4484) added to known issues. Please review

.{ProductShortName} CLI bulk analysis does not function on Windows

{ProductShortName} CLI bulk analysis, performed by using the `--bulk` option, does not function on Microsoft Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4307[(MTA-4307)]

.{ProductShortName} Containerless CLI does not fire XML custom rules

The {ProductShortName} Containerless CLI does not fire XML custom rules, and the `analyze-bin` command is not raising violations coming from custom rules in XML format. link:https://issues.redhat.com/browse/MTA-4097[(MTA-4097)

.Binary analysis fails with incorrect Maven credentials

{ProductShortName} CLI analysis fails with a generic error when incorrect Maven credentials are used. The failure occurs because the analyzer is unable to report detailed error messages. link:https://issues.redhat.com/browse/MTA-3714[(MTA-3714)]

> .{ProductShortName} CLI returns confusing error messages when Python is not installed on Windows

Author: @ibragins - FYI

{ProductShortName} CLI returns two confusing error messages of `failed to validate flags` and `exit status 9009` when Python is not installed on Microsoft Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4319[(MTA-4319)]

> .{ProductShortName} Containerless CLI Java remains in memory after analysis is completed

Author: @ibragins - FYI

Reviewer 3: @ibragins I'm not sure why the release notes label was set here; this bug has been verified and this is not an issue that will be included in the release. @anarnold97 please remove.

:_template-generated: 2024-12-04
:_mod-docs-content-type: REFERENCE

[id="known-issues-7-2-0_{context}"]
= Known issues

This section provides highlighted known issues in {ProductFullName} version 7.2.0.

> .{ProductShortName} [CLI] analysis using the Container CLI fails on Windows

Reviewer 3: I'd prefer `analysis using the CLI in container mode periodically fails on windows`

> {ProductShortName} [CLI] analysis using the Container CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]

Reviewer 3: NIT:
>This failure can be caused by the existence of a corrupted `.metadata` folder located inside the directory where analysis is run

> .{ProductShortName} [CLI] analysis using the Container CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside of the folder where the analysis is run. Threfore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]

Author: ```suggestion
{ProductShortName} [CLI] analysis using the Container CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]
```

.{ProductShortName} [CLI] analysis using the Containerless CLI fails on Windows

> {ProductShortName} [CLI] analysis using the Containerless CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4416[(MTA-4416)]

Author: @ibragins && @dymurray - please can I flag these two issues to you for the following two tickets:

    * [(MTA-4416)](https://issues.redhat.com/browse/MTA-4416)
    * [(MTA-4410)](https://issues.redhat.com/browse/MTA-4410)

Do you want a similar admonition added to the CLI documentation?

Thanks

Reviewer 3: I think these are just 2 sections duplicated right? You can just delete one of them and the comments I left above are good for me. One of these tickets were closed

> .{ProductShortName} [CLI] analysis using the Container CLI fails on Windows
> 
> {ProductShortName} [CLI] analysis using the Container CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]
> 
> .{ProductShortName} [CLI] analysis using the Containerless CLI fails on Windows
> 
> {ProductShortName} [CLI] analysis using the Containerless CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4416[(MTA-4416)]

Author: ```suggestion
.{ProductShortName} CLI analysis using the Container CLI  and the Containerless CLI fails on Windows

{ProductShortName} [CLI] analysis using the Container CLI and the Containerless CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4410[(MTA-4410)]
```

Suggestion following Igor's comment

> .{ProductShortName} [CLI] analysis using the Containerless CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside of the folder where the analysis is run. Threfore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4416[(MTA-4416)]

Author: ```suggestion
{ProductShortName} [CLI] analysis using the Containerless CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside the folder where the analysis is run. Therefore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4416[(MTA-4416)]
```

> .{ProductShortName} [CLI] analysis using the Containerless CLI fails on Microsoft Windows and returns a `provider does not have dependency capability` error. This failure is caused by the existence of the `.metadata` folder located inside of the folder where the analysis is run. Threfore, the workaround is to delete this folder. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4416[(MTA-4416)]

Author: @ibragins && @dymurray - please can I flag these two issues to you for 

* [(MTA-4416)](https://issues.redhat.com/browse/MTA-4416)
* [(MTA-4410)](https://issues.redhat.com/browse/MTA-4410)

Thanks

.{ProductShortName} CLI analysis is extremely slow on Windows

{ProductShortName} CLI analysis does not function as expected on Microsoft Windows, with analysis using container runtimes being extremely slow on Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-3667[(MTA-3667)]

> .Analysis of the daytrader application does not show dependencies

Reviewer 3: This shouldn't mention the daytrader application.

At the root of it; this issue that is linked is about how when running analysis in containerless mode the CLI will be using the version of maven that exists on the host. We have noticed that analysis with a host that has maven 3.6 installed sometimes it will fail to fetch some dependencies.

The solution is for the user to upgrade to maven 3.9

The analysis of the daytrader application does not show dependencies. This failure to display dependencies is caused by the {ProductShortName} Operator using Maven 3.6, which sometimes fails to fetch dependencies. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4169[(MTA-4169)]

.{ProductShortName} CLI bulk analysis does not function on Windows

{ProductShortName} CLI bulk analysis, performed by using the `--bulk` option, does not function on Microsoft Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4307[(MTA-4307)]

> .{ProductShortName} Containerless CLI does not fire XML custom rules

Reviewer 3: @ibragins this bug has been verified as well right? I think this section can be removed...

The {ProductShortName} Containerless CLI does not fire XML custom rules, and the `analyze-bin` command is not raising violations coming from custom rules in XML format. link:https://issues.redhat.com/browse/MTA-4097[(MTA-4097)

.Binary analysis fails with incorrect Maven credentials

{ProductShortName} CLI analysis fails with a generic error when incorrect Maven credentials are used. The failure occurs because the analyzer is unable to report detailed error messages. link:https://issues.redhat.com/browse/MTA-3714[(MTA-3714)]

.{ProductShortName} CLI returns confusing error messages when Python is not installed on Windows

> {ProductShortName} CLI returns two confusing error messages of `failed to validate flags` and `exit status 9009` when Python is not installed on Microsoft Windows. This issue is planned to be resolved in {ProductShortName} 7.2.1. link:https://issues.redhat.com/browse/MTA-4319[(MTA-4319)]

Reviewer 3: I apologize; I'm not sure why this bug was still open with a release notes label. I just closed it as it's obsolete. You can remove this section entirely
